### ISSUE TITLE
trouble building HP

### ISSUE THREAD

**Issue by erantapaa** [haskell/haskell-platform#241]
I'm having trouble building the current HP (commit 21e1fa8c...)

A synopsis of my setup:

- a clean Ubuntu 14.04.4 x64 system
- installed the previous Haskell Platform - 7.10.3
- installed the most current stack
- using the GHC bindist ghc-8.0.1-x86_64-deb8-linux.tar.xz
- `hptool` builds OK
- Ran the command:

```
./platform.sh ../ghc-8.0.1-x86_64-deb8-linux.tar.xz /usr/local/bin/{cabal,stack} build-all
```

When hptool tries to build `hscolour-1.24` I'm getting:

```
...
on the commandline: warning:
    -this-package-key is deprecated: Use -this-unit-id instead
cabal:
'/home/erantapaa/build/haskell-platform/build/ghc-bindist/local/bin/ghc-pkg'
exited with an error:
hscolour-1.24: Warning: haddock-interfaces:
/home/erantapaa/build/haskell-platform/build/package/hscolour-1.24/build/dist/doc/html/hscolour/hscolour.haddock
doesn't exist or isn't a file
hscolour-1.24: Warning: haddock-html:
/home/erantapaa/build/haskell-platform/build/package/hscolour-1.24/build/dist/doc/html/hscolour
doesn't exist or isn't a directory
hscolour-1.24: installed package info from too old version of Cabal (key field
does not match id field)
)
```

and the build aborts.

Before that I see these lines:

```
In-place registering hscolour-1.24...
cabal:
'/home/erantapaa/build/haskell-platform/build/ghc-bindist/local/bin/ghc-pkg'
exited with an error:
hscolour-1.24: Warning: haddock-interfaces:
/home/erantapaa/build/haskell-platform/build/package/hscolour-1.24/build/dist/doc/html/hscolour/hscolour.haddock
doesn't exist or isn't a file
hscolour-1.24: Warning: haddock-html:
/home/erantapaa/build/haskell-platform/build/package/hscolour-1.24/build/dist/doc/html/hscolour
doesn't exist or isn't a directory
hscolour-1.24: installed package info from too old version of Cabal (key field
does not match id field)
Error when running Shake build system:
* build-all
* build-product
* build/product/haskell-platform-8.0.1-unknown-posix--minimal-x86_64.tar.gz
* build/product/hp-usr-local.tar.gz
* build/target
* build/.markers/rel/build/target/usr/local/haskell/ghc-8.0.1-x86_64/lib/hscolour-1.24
* build/.markers/rel/build/package/hscolour-1.24/build
user error (Development.Shake.command, system command failed
Command: cabal build --verbose=1
Current directory: /home/erantapaa/build/haskell-platform/build/package/hscolour-1.24/build
Exit code: 1
```

Any ideas?

**Comment by gbaz** [comment]
Erik — you’ll need to use the latest released cabal (1.24) as your system cabal.

**Comment by erantapaa** [comment]
Ok - that fixed the problem - thanks!
